changedetectionio/model/Watch.py

```diff
@@ -178,3 +213,25 @@ def lines_contain_something_unique_compared_to_history(self, lines=[]):
                 return True
 
         return False
+
+    # Get diff types (addition, deletion, modification) from the previous snapshot and new_text
+    # uses similar algorithm to customSequenceMatcher in diff.py
+    # Returns a dict of diff types and wether they are present in the diff
+    def get_diff_types(self, new_text):
+        import difflib
+
+        diff_types = {
+            'add': False,
+            'del': False,
+        }
+
+        # get diff types using difflib
+        cruncher = difflib.SequenceMatcher(isjunk=lambda x: x in " \\t", a=str(self.get_previous_text()), b=str(new_text))
```

Comment:
Does this also ignore line-feeds?

Reply:
what do you mean by line feeds?

Reply:
`isjunk=lambda x: x in " \\t"` so what if a page adds a line-feed, should that get included as a change?

Reply:
I am using the same algorithm that is used for notification generation. I think that only if text is added that a change should be considered?